File: workspace/src/controllers/TaskController.js
// src/controllers/TaskController.js
import { ref } from 'vue';

export default function useTaskController() {
  const tasks = ref([]);
  const loading = ref(false);
  const error = ref(null);

  const fetchTasks = async () => {
    loading.value = true;
    error.value = null;
    try {
      const res = await fetch('https://jsonplaceholder.typicode.com/todos?_limit=5');
      if (!res.ok) throw new Error('Error al cargar tareas');
      tasks.value = await res.json();
    } catch (err) {
      error.value = err.message;
    } finally {
      loading.value = false;
    }
  };

  const addTask = (title) => {
    if (!title.trim()) return;
    tasks.value.unshift({ id: Date.now(), title, completed: false });
  };

  const toggleTask = (id) => {
    tasks.value = tasks.value.map(t => t.id === id ? { ...t, completed: !t.completed } : t);
  };

  const deleteTask = (id) => {
    tasks.value = tasks.value.filter(t => t.id !== id);
  };

  return { tasks, loading, error, fetchTasks, addTask, toggleTask, deleteTask };
}

File: workspace/backend/task_controller.py
import requests

class TaskController:
    def __init__(self):
        self.tasks = []
        self.loading = False
        self.error = None

    def fetch_tasks(self):
        self.loading = True
        self.error = None
        try:
            res = requests.get('https://jsonplaceholder.typicode.com/todos?_limit=5')
            res.raise_for_status()
            self.tasks = res.json()
        except Exception as err:
            self.error = str(err)
        finally:
            self.loading = False
        return self.tasks

    def add_task(self, title):
        if not title.strip():
            return
        self.tasks.insert(0, {'id': id(self), 'title': title, 'completed': False})

    def toggle_task(self, id):
        for t in self.tasks:
            if t['id'] == id:
                t['completed'] = not t['completed']

    def delete_task(self, id):
        self.tasks = [t for t in self.tasks if t['id'] != id]
